PurchasesCoreSwift/Networking/Backend.swift

```diff
@@ -0,0 +1,614 @@
+
+            let attributesInBackendFormat = subscriberAttributesToDict(subscriberAttributes: subscriberAttributesByKey)
+            body["attributes"] = attributesInBackendFormat as AnyObject
+        }
+
+        if let offeringIdentifier = offeringIdentifier {
+            body["presented_offering_identifier"] = offeringIdentifier as NSString
+        }
+
+        httpClient.performPOSTRequest(serially: true,
+                                      path: "/receipts",
+                                      requestBody: body,
+                                      headers: authHeaders) { statusCode, response, error in
+            let callbacks = self.getPurchaserInfoCallbacksAndClearCache(forKey: cacheKey)
+            for callback in callbacks {
+                self.handle(purchaserInfoResponse: response, statusCode: statusCode, error: error, completion: callback)
+            }
+        }
+    }
+
+    @objc
+    public func getSubscriberData(appUserID: String, completion: @escaping BackendPurchaserInfoResponseHandler) {
+        let escapedAppUserID = escapedAppUserID(appUserID: appUserID)
+        let path = "/subscribers/\(escapedAppUserID)"
+
+        if add(callback: completion, key: path) {
+            return
+        }
+
+        httpClient.performGETRequest(serially: true, path: path, headers: authHeaders) { statusCode, response, error in
+            for completion in self.getPurchaserInfoCallbacksAndClearCache(forKey: path) {
+                print("Hi")
+                self.handle(purchaserInfoResponse: response,
+                            statusCode: statusCode,
+                            error: error,
+                            completion: completion)
+            }
+        }
+    }
+
+    // swiftlint:disable function_parameter_count
+    @objc(postOfferForSigning:withProductIdentifier:subscriptionGroup:receiptData:appUserID:completion:)
+    public func post(offerIdForSigning offerIdentifier: String,
+                     productIdentifier: String,
+                     subscriptionGroup: String,
+                     receiptData: Data,
+                     appUserID: String,
+                     completion: @escaping OfferSigningResponseHandler) {
+    // swiftlint:enable function_parameter_count
+        let fetchToken = receiptData.base64EncodedString(options: .lineLength64Characters)
+
+        let requestBody: [String: Any] = ["app_user_id": appUserID,
+                                          "fetch_token": fetchToken,
+                                          "generate_offers": [
+                                            ["offer_id": offerIdentifier,
+                                             "product_id": productIdentifier,
+                                             "subscription_group": subscriptionGroup
+                                            ]
+                                          ]]
+
+        self.httpClient.performPOSTRequest(serially: true, path: "/offers",
```

Comment:
[Style] `path` should be in another line, shouldn't be?

Reply:
Oops